packages/wonder-blocks-data/src/util/__tests__/response-cache.test.js

```diff
@@ -436,105 +459,371 @@ describe("../response-cache.js", () => {
+                    const cache = new ResponseCache(hydrationCache);
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: true,
+                    };
 
-            // Assert
-            expect(result).toStrictEqual({data: "data!"});
+                    // Act
+                    const result = cache.getEntry(fakeHandler, "options");
+
+                    // Assert
+                    expect(result).toBeNull();
+                });
+
+                it("should return the cached entry if in the hydration cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    jest.spyOn(hydrationCache, "retrieve").mockReturnValue({
+                        data: "data!",
+                    });
+                    const cache = new ResponseCache(hydrationCache);
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: true,
+                    };
+
+                    // Act
+                    const result = cache.getEntry(fakeHandler, "options");
+
+                    // Assert
+                    expect(result).toStrictEqual({data: "data!"});
+                });
+            });
+
+            describe("handler does not want hyrdation", () => {
+                it("should return null if not in the ssr-only cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    const ssrOnlyCache = new MemoryCache();
+                    jest.spyOn(ssrOnlyCache, "retrieve").mockReturnValue(null);
+                    const cache = new ResponseCache(
+                        hydrationCache,
+                        ssrOnlyCache,
+                    );
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: true,
+                    };
+
+                    // Act
+                    const result = cache.getEntry(fakeHandler, "options");
+
+                    // Assert
+                    expect(result).toBeNull();
+                });
+
+                it("should return the cached entry if in the hydration cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    const ssrOnlyCache = new MemoryCache();
+                    jest.spyOn(ssrOnlyCache, "retrieve").mockReturnValue({
+                        data: "data!",
+                    });
```

Comment:
Is the title correct for this test?  It looks like the `ssrOnlyCache` is the one with the data.

Reply:
I'll check

Reply:
Good catch. Thanks